### ISSUE TITLE
upgrade to latest pushy

### ISSUE THREAD

**Issue by divs1210** [divs1210/pushy-clj#6]
Hi, @brunchboy!
I noticed there have been a lot of enhancements and bugfixes in pushy.
I've pushed a patch under a new branch, and deployed a 0.5.0-SNAPSHOT release.
I don't have access to a developer account at the moment.
Would it be possible for you to test it out?

**Comment by brunchboy** [comment]
Interesting, thanks for letting me know. I will try to find some time to bring up that part of our stack on my development machine next week and do some testing. Is there anything to focus on?

**Comment by divs1210** [comment]
Check out [this release](https://github.com/relayrides/pushy/releases/tag/pushy-0.13.0).

**Comment by brunchboy** [comment]
I didn’t have time to test this during the week, and am about to leave for some international travel, but my colleague Marc will hopefully be able to try it tomorrow or Monday, and I’ll be back after that.

**Comment by divs1210** [comment]
Thank you!

**Comment by rboyd** [comment]
I'm just randomly stumbling in here to say everything wfm except I can only directly:

```clojure
(.sendNotification client notification)
```
instead of
```clojure
(send-push-notification client notification)
```

which throws:
```

1. Unhandled java.lang.NoSuchMethodError
   com.turo.pushy.apns.ApnsClient.sendNotification(Lcom/turo/pushy/apns/ApnsPushNotification;)Lio/netty/util/concurrent/Future;

                  core.clj:   92  pushy-clj.core/send-push-notification
                  core.clj:   84  pushy-clj.core/send-push-notification
                      REPL:   51  etp-serv.push/eval4521
                      REPL:   51  etp-serv.push/eval4521
             Compiler.java: 7062  clojure.lang.Compiler/eval
             Compiler.java: 7025  clojure.lang.Compiler/eval
                  core.clj: 3206  clojure.core/eval
                  core.clj: 3202  clojure.core/eval
                  main.clj:  243  clojure.main/repl/read-eval-print/fn
                  main.clj:  243  clojure.main/repl/read-eval-print
                  main.clj:  261  clojure.main/repl/fn
                  main.clj:  261  clojure.main/repl
                  main.clj:  177  clojure.main/repl
               RestFn.java: 1523  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   87  clojure.tools.nrepl.middleware.interruptible-eval/evaluate/fn
                  AFn.java:  152  clojure.lang.AFn/applyToHelper
                  AFn.java:  144  clojure.lang.AFn/applyTo
                  core.clj:  657  clojure.core/apply
                  core.clj: 1965  clojure.core/with-bindings*
                  core.clj: 1965  clojure.core/with-bindings*
               RestFn.java:  425  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   85  clojure.tools.nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:   55  clojure.tools.nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:  222  clojure.tools.nrepl.middleware.interruptible-eval/interruptible-eval/fn/fn
    interruptible_eval.clj:  190  clojure.tools.nrepl.middleware.interruptible-eval/run-next/fn
                  AFn.java:   22  clojure.lang.AFn/run
   ThreadPoolExecutor.java: 1149  java.util.concurrent.ThreadPoolExecutor/runWorker
   ThreadPoolExecutor.java:  624  java.util.concurrent.ThreadPoolExecutor$Worker/run
               Thread.java:  748  java.lang.Thread/run

```
Do we need to update type hints maybe?

**Comment by divs1210** [comment]
Hi, @rboyd!
Thanks for testing!
Can you try forking and modifying the code?
Like I said, I don't have access to a dev account atm.
A patch will be highly appreciated 😄

**Comment by divs1210** [comment]
Hi @brunchboy, @rboyd  could you please test with `0.5.1-SNAPSHOT`?

**Comment by brunchboy** [comment]
We just tested 0.5.0-SNAPSHOT on Friday and it seemed to work in a quick smoke test. We can try to find some more time this coming week.

**Comment by lensgolda** [comment]
Please update, got another error
```
IllegalStateException Channel closed before HTTP/2 preface completed.  com.turo.pushy.apns.ApnsChannelFactory$3$2.operationComplete (ApnsChannelFactory.java:198)


WARN io.netty.channel.ChannelInitializer - Failed to initialize a channel. Closing: [id: 0xcbde87ef]
java.lang.NoSuchFieldError: SSL_MAX_RECORD_LENGTH
        at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.<clinit>(ReferenceCountedOpenSslEngine.java:129)
        at io.netty.handler.ssl.ReferenceCountedOpenSslContext.newEngine0(ReferenceCountedOpenSslContext.java:372)
        at io.netty.handler.ssl.ReferenceCountedOpenSslContext.newHandler(ReferenceCountedOpenSslContext.java:363)
        at io.netty.handler.ssl.SslContext.newHandler(SslContext.java:905)
        at com.turo.pushy.apns.ApnsChannelFactory$1.initChannel(ApnsChannelFactory.java:98)
        at com.turo.pushy.apns.ApnsChannelFactory$1.initChannel(ApnsChannelFactory.java:88)
        at io.netty.channel.ChannelInitializer.initChannel(ChannelInitializer.java:113)
        at io.netty.channel.ChannelInitializer.handlerAdded(ChannelInitializer.java:105)
        at io.netty.channel.DefaultChannelPipeline.callHandlerAdded0(DefaultChannelPipeline.java:617)
        at io.netty.channel.DefaultChannelPipeline.access$000(DefaultChannelPipeline.java:46)
        at io.netty.channel.DefaultChannelPipeline$PendingHandlerAddedTask.execute(DefaultChannelPipeline.java:1467)
        at io.netty.channel.DefaultChannelPipeline.callHandlerAddedForAllHandlers(DefaultChannelPipeline.java:1141)
        at io.netty.channel.DefaultChannelPipeline.invokeHandlerAddedIfNeeded(DefaultChannelPipeline.java:666)
        at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:510)
        at io.netty.channel.AbstractChannel$AbstractUnsafe.register(AbstractChannel.java:476)
        at io.netty.channel.SingleThreadEventLoop.register(SingleThreadEventLoop.java:80)
        at io.netty.channel.SingleThreadEventLoop.register(SingleThreadEventLoop.java:74)
        at io.netty.channel.MultithreadEventLoopGroup.register(MultithreadEventLoopGroup.java:86)
        at io.netty.bootstrap.AbstractBootstrap.initAndRegister(AbstractBootstrap.java:333)
        at io.netty.bootstrap.Bootstrap.doResolveAndConnect(Bootstrap.java:163)
        at io.netty.bootstrap.Bootstrap.connect(Bootstrap.java:119)
        at com.turo.pushy.apns.ApnsChannelFactory$3.run(ApnsChannelFactory.java:176)
        at io.netty.util.concurrent.PromiseTask$RunnableAdapter.call(PromiseTask.java:38)
        at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:125)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:886)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.lang.Thread.run(Thread.java:748)
```

seems like that happened because of netty dependencie, in old version of pushy.

**Comment by divs1210** [comment]
Hi @lensgolda, which version of pushy-clj did you test with?

**Comment by lensgolda** [comment]
Hi, that trace occurs with latest stable `[pushy-clj "0.4.0"]` for me

**Comment by ezcocoa** [comment]
Hi @lensgolda. i got same error. the problem is dependency library that is ' [com.turo/pushy "0.11.3"]'. 
i've changed the version from '0.11.3' to '0.13.3'. it's worked well.

ref: https://github.com/relayrides/pushy

**Comment by divs1210** [comment]
Hi @lensgolda, I've released `0.4.1` with @hjbaek's suggested change.
Please see #7.
Thank you for your contribution!

**Comment by divs1210** [comment]
@lensgolda, @hjbaek are you also seeing the error that @rboyd mentioned?

**Comment by divs1210** [comment]
Oops 😞
I was working on 0.5.1 for this upgrade.
I panicked because it seems like 0.4.0 has a problem, and hence released 0.4.1 with updated pushy, but without many of the other changes in 0.5.1.

**Comment by ezcocoa** [comment]
@divs1210 Nope. i just got the error that @lensgolda mentioned. it was not the error that @rboyd mentioned.

**Comment by divs1210** [comment]
@hjbaek thanks for the information!

**Comment by lensgolda** [comment]
I mention about my error, to this issue, because reason i got it, as i supposed here in my message

> seems like that happened because of netty dependencie, in **old** version of pushy.

, was (very) old version of pushy (java library), so @hjbaek was right and confirm that

> changed the version from '0.1**1**.3' to '0.1**3**.3'. it's worked well

so all problems must gone, because you did exactly what i meant

> upgrade to latest pushy

as this issue was named.

Thank's anyway ;)

**Comment by brunchboy** [comment]
It seems that 0.5.1 was never released? Are you no longer looking for testing, though?

**Comment by divs1210** [comment]
Hi, @brunchboy!
Sorry I didn't reply sooner.
I released 0.4.1 with some of the changes that were supposed to be a part of 0.5.x.
I certainly am looking for testing, but am unable to give much time to this atm.
There were some minor code cleanups that were also going to be in 0.5.x that are not a part of 0.4.1.
Feel free to test/send PRs, develop/maintain the lib further.
